.circleci/config.yml

```diff
@@ -302,6 +309,7 @@ jobs:
     branches:
     steps:
       - checkout
+      - git-secrets-check
```

Comment:
nit: It looks like this will scan the whole repository but only runs during test-backend, which may be fine but I could see it being confusing if the file with secrets is in the frontend but we are failing backend tests. Additionally, there's stuff outside of the backend that we would want to check as well. Do you think it makes sense to make this its own job and call it as a distinct step in the workflow that is a pre-requisite to both `test-backend` and `test-frontend`?

Reply:
@abottoms-coder can you check @jtwillis92's comment above ☝🏾 and let me know how you want to handle? i agree that it would be a good idea to run the check beyond just the backend (probably anywhere where code lives).

Reply:
also, the ci piece could be addressed in #966

Reply:
Ok, I believe I have implemented what John described here. I think we might need to pause pending his review that I captured his intent.